worker/uniter/operation/executor.go

```diff
@@ -70,9 +71,22 @@ func (x *executor) Run(op Operation) error {
 	switch err := x.do(op, stepPrepare); errors.Cause(err) {
 	case ErrSkipExecute:
 	case nil:
+		done := make(chan struct{})
+		go func() {
+			for {
+				select {
+				case rs := <-remoteStateChange:
```

Comment:
Need a check for channel closing?
```
rs, ok := <-remoteStateChange:
if !ok {
  return
}
```

Reply:
the channel never closes but I can do that in case the semantics change